pkg/manager/signals/signal.go

```diff
@@ -26,18 +27,19 @@ var onlyOneSignalHandler = make(chan struct{})
 // SetupSignalHandler registers for SIGTERM and SIGINT. A stop channel is returned
 // which is closed on one of these signals. If a second signal is caught, the program
 // is terminated with exit code 1.
-func SetupSignalHandler() (stopCh <-chan struct{}) {
+func SetupSignalHandler() (ctx context.Context) {
```

Comment:
We pass this directly to `Start` in multiple places, which now accepts a context

Reply:
In a way jason has a good point though, I think the reason the context package has the `don't store it in a struct` comment is because a context is not only for canceling but also to transport request-scoped information. But we don't do that here, we only care about the canceling aspect, so the channel might actually a better fit than the context?

Reply:
Yeah, wasn't objecting to it, just wanted to point out that we're accepting a context now. In a way, it works like stdlib behaves in a lot of places (http server is an example) so I copied that approach

Reply:
Fixed, thank you all